### ISSUE TITLE
Card browser freezes after using "Search all decks" button

### ISSUE THREAD

**Issue by Anthropos888** [ankidroid/Anki-Android#6500]
###### Reproduction Steps

1. have 2 or more decks, review a card from one of the decks
2. long-tap the word on the front -> choose "card browser" from the floating menu
3. the card browser search result from the current deck is shown -> tap "search all decks"
4. tap to open any of the search results -> press your phone's back button to return to the card browser

###### Expected Result

search results are shown

###### Actual Result

cards browser is empty and shows the reload icon in the middle. Nothing happens when tapping the reload button. No further interaction with card browser possible. Only the phone's back button can exit the card browser.

###### Debug info
AnkiDroid Version = 2.12alpha10

Android Version = 9

ACRA UUID = 1f29f610-893a-403f-83d5-25a26ba97fe0

###### Research
*Enter an [x] character to confirm the points below:*

- [ x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x ] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x ] I have searched for similar existing issues here and on the user forum
- [x ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by david-allison-1** [comment]
I managed to reproduce this initially. I'm now struggling to reproduce this.

It's not due to #6496 / dff3614215a5a11da63a6747b4148a832a5b86b0 as pressing "back" won't return an intent.

**Comment by Anthropos888** [comment]
@david-allison-1 I can reproduce it with the given steps over and over again. The important thing: after you reproduced it once, you need to exit the reviewer, go back to the deck list and start reviewing the deck again in order to reproduce it again. 

If you just return to the reviewer for the current deck after pressing "search all decks", the deck selection in the card browser will remember "all decks". Thus the bug won't occur anymore when selecting the word on the front > card browser > open the first entry > go back to the search results. Only after going back to the deck list the bug is reproducable again.

**Comment by Anthropos888** [comment]
Actually the card browser doesn't freeze, because I can still select the columns and choose a specific deck from the deck selection. 
And after choosing another deck, the search results are shown and the search all decks button appears again.

Clicking on the search all decks button once more will reproduce the bug again.

**Comment by Arthur-Milchior** [comment]
Do you have any idea how many cards the search would return if he was sucesful ? You can probably search on anki desktop to get an idea.

I've not yet opened the profiler, but my intuition is that your smartphone is spending a lot of time computing the content of the result instead of displaying it. If I'm right, card should appear if you wait long enough. However, I've no idea how "long" is "long enough"

**Comment by Anthropos888** [comment]
@Arthur-Milchior I just tried it with a search that only returns 2 cards. When tapping one of the cards and returning to the card browser with the phone's back button, the card browser is empty, only showing the reload icon.
I'm waiting for 3 minutes now, nothing happens.

**Comment by david-allison-1** [comment]
Reproduced again, thanks. Will get on this today.

**Comment by Arthur-Milchior** [comment]
In this case, I can't reproduce it at all. I did the search on my collection, with the word "there" so a lot of answer. It takes 20 seconds before I see the first answer. A lot of this time is spent in the hashmap method and transforming numbers into string. However, there is already a PR solving this innefficiency. I'm a little bit sad to discover that this PR won't solve the current problem

**Comment by david-allison-1** [comment]
Debug log: [6500.txt](https://github.com/ankidroid/Anki-Android/files/4805368/6500.txt). 

Can't see anything initially, will add more logging.

**Comment by david-allison-1** [comment]
This got even more complicated. It's visible in the "flicker" of the top bar from "search" to "select a deck" to "search".

``onItemSelected` is caused by `supportInvalidateOptionsMenu`. 

Turns out we hid the "search" when we did an `onCreateOptionsMenu`, sometimes this triggered UI actions (such as searches). As the search was performed just before we changed screens, it was cancelled, but the progress dialog was not hidden
